rust/cardano-chain-follower/src/follow.rs

```diff
@@ -141,16 +142,21 @@ impl ChainFollower {
 
             if let Some(follower) = self.mithril_follower.as_mut() {
                 if let Some(next) = follower.next().await {
-                    self.previous = self.current.clone();
-                    self.current = next.point();
-                    self.fork = Fork::IMMUTABLE; // Mithril Immutable data is always Fork 0.
                     let update = ChainUpdate::new(chain_update::Kind::Block, false, next);
                     return Some(update);
                 }
+
+                // Verifying ultra rare scenario of race condition on the mithril snapshot data
+                // directory, where the underlying data directory could be no longer accessible
+                if !follower.is_valid() {
```

Comment:
For me this logic is weird.
I think it would be better like:
```
if follower.is_vaild { break }
```

Remove the `return None` at line 158 and re-intrioduce the `None` at 153.
Then the code doing the recovery doesn't need `continue`.